libapicore/ApiServer.cpp

```diff
@@ -645,15 +645,15 @@ void ApiConnection::processRequest(Json::Value& jRequest, Json::Value& jResponse
             }
             else
             {
-                if (!getRequestValue("noncescrambler", nonce, jRequestParams, true, jResponse))
+                // as we already know there is a "noncescrambler" element we can use optional=false
+                if (!getRequestValue("noncescrambler", nonce, jRequestParams, false, jResponse))
```

Comment:
I think here we should also add
```
                    jResponse["error"]["code"] = -422;
                    jResponse["error"]["message"] = "Invalid nonce";
```
before the return

Otherwise we return an incomplete response to issuer

Reply:
Invalid value error is done within 'getRequestValue()'